pkg/rest/apisurface.go

```diff
@@ -265,3 +265,45 @@ func writeError(w http.ResponseWriter, err error) {
 
 	writeErrorResponse(w, http.StatusInternalServerError, err)
 }
+
+// UpdateHandler is the mux handler that dispatches Update requests to the
+// broker's BusinessLogic.
+func (s *APISurface) UpdateHandler(w http.ResponseWriter, r *http.Request) {
+	version := getBrokerAPIVersionFromRequest(r)
+	if err := s.BusinessLogic.ValidateBrokerAPIVersion(version); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	request, err := unpackUpdateRequest(r)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	glog.Infof("Received Update Request for instanceID %q", request.InstanceID)
+
+	response, err := s.BusinessLogic.Update(request, w, r)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	status := http.StatusOK
+	if response.Async {
+		status = http.StatusAccepted
+	}
+
+	writeResponse(w, status, response)
+}
+
+func unpackUpdateRequest(r *http.Request) (*osb.UpdateInstanceRequest, error) {
+	osbRequest := &osb.UpdateInstanceRequest{}
+
+	vars := mux.Vars(r)
+	osbRequest.ServiceID = vars[serviceIDVarKey]
+	planID := vars[planIDVarKey]
```

Comment:
This has to be done as the osb-client expects a string pointer. https://github.com/pmorie/go-open-service-broker-client/blob/master/v2/update_instance.go#L12

I am assuming that was done to indicate the PlanID is not mandatory as per the spec?

Reply:
That's correct - the planID is sent, according to the spec, only when the user explicitly requests that the plan should change.  See: https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#body-4

So, you won't always get it.  I'm thinking that if it isn't sent to you, shouldn't get a pointer to an empty string.  Let's check the value of the var and if it's empty, avoid setting the pointer.  WDYT?